zipline/finance/blotter.py

```diff
@@ -179,64 +185,62 @@ def process_benchmark(self, benchmark_event):
+        there were no open orders, an empty list is returned.
+        """
+        closed_orders = []
+        transactions = []
+
+        for asset, asset_orders in self.open_orders.iteritems():
+            for order, txn in self.slippage_func(asset_orders, self.current_dt):
+                direction = math.copysign(1, txn.amount)
+                per_share, total_commission = self.commission.calculate(txn)
+                txn.price += per_share * direction
+                txn.commission = total_commission
+                order.filled += txn.amount
 
-        if trade_event.sid not in self.open_orders:
-            return
+                if txn.commission is not None:
+                    order.commission = (order.commission or 0.0) + \
+                                       txn.commission
 
-        if trade_event.volume < 1:
-            # there are zero volume trade_events bc some stocks trade
-            # less frequently than once per minute.
-            return
+                txn.dt = pd.Timestamp(txn.dt, tz='UTC')
+                order.dt = txn.dt
+
+                transactions.append(txn)
+
+                if not order.open:
+                    closed_orders.append(order)
 
-        orders = self.open_orders[trade_event.sid]
-        orders.sort(key=lambda o: o.dt)
-        # Only use orders for the current day or before
-        current_orders = filter(
-            lambda o: o.dt <= trade_event.dt,
-            orders)
-
-        processed_orders = []
-        for txn, order in self.process_transactions(trade_event,
-                                                    current_orders):
-            processed_orders.append(order)
-            yield txn, order
-
-        # remove closed orders. we should only have to check
-        # processed orders
-        def not_open(order):
-            return not order.open
-        closed_orders = filter(not_open, processed_orders)
         for order in closed_orders:
-            orders.remove(order)
-
-        if len(orders) == 0:
-            del self.open_orders[trade_event.sid]
-
-    def process_transactions(self, trade_event, current_orders):
-        for order, txn in self.transact(trade_event, current_orders):
-            if txn.type == zp.DATASOURCE_TYPE.COMMISSION:
-                order.commission = (order.commission or 0.0) + txn.cost
-            else:
-                if txn.amount == 0:
-                    raise zipline.errors.TransactionWithNoAmount(txn=txn)
-                if math.copysign(1, txn.amount) != order.direction:
-                    raise zipline.errors.TransactionWithWrongDirection(
-                        txn=txn, order=order)
-                if abs(txn.amount) > abs(self.orders[txn.order_id].amount):
-                    raise zipline.errors.TransactionVolumeExceedsOrder(
-                        txn=txn, order=order)
+            sid = order.sid
+            if sid not in self.open_orders:
```

Comment:
To reduce the number of calls this could be written as:

```
try:
     sid_orders = self.open_orders[sid]
     sid_orders.remove(order)
except KeyError:
      continue
```

Is self.open_orders[sid] a list? Something that we want to do targeted removal on may work better as a `dict` keyed by the order.id